plugin/pkg/seed/validator/admission.go

```diff
@@ -191,22 +217,93 @@ func getOldAndNewSeeds(attrs admission.Attributes) (*core.Seed, *core.Seed, erro
-		return nil
+	var (
+		backup              = newSeed.Spec.Backup
+		getAttributesRecord = func(ref *corev1.ObjectReference) (authorizer.AttributesRecord, error) {
+			var (
+				apiGroup   string
+				apiVersion string
+				resource   string
+			)
+			if ref.APIVersion == corev1.SchemeGroupVersion.String() {
+				apiGroup = corev1.SchemeGroupVersion.Group
+				apiVersion = corev1.SchemeGroupVersion.Version
+				resource = "secrets"
+			} else if ref.APIVersion == securityv1alpha1.SchemeGroupVersion.String() {
+				apiGroup = securityv1alpha1.SchemeGroupVersion.Group
+				apiVersion = securityv1alpha1.SchemeGroupVersion.Version
+				resource = "workloadidentities"
+			} else {
+				return authorizer.AttributesRecord{}, errors.New("unsupported credentials reference: backup config is referencing neither a Secret nor a WorkloadIdentity")
+			}
+			return authorizer.AttributesRecord{
+				User:            attrs.GetUserInfo(),
+				Verb:            "get",
+				APIGroup:        apiGroup,
+				APIVersion:      apiVersion,
+				Resource:        resource,
+				Namespace:       ref.Namespace,
+				Name:            ref.Name,
+				ResourceRequest: true,
+			}, nil
+		}
+	)
+
+	switch {
+	case backup.CredentialsRef.APIVersion == securityv1alpha1.SchemeGroupVersion.String() &&
+		backup.CredentialsRef.Kind == "WorkloadIdentity":
+		workloadIdentity, err := v.workloadIdentityLister.WorkloadIdentities(backup.CredentialsRef.Namespace).Get(backup.CredentialsRef.Name)
+		if err != nil {
+			return apierrors.NewInternalError(err)
+		}
+
+		if seedBackupType, workloadIdentityType := backup.Provider, workloadIdentity.Spec.TargetSystem.Type; seedBackupType != workloadIdentityType {
+			return admission.NewForbidden(attrs, fmt.Errorf("seed using backup of type %q cannot use WorkloadIdentity of type %q", seedBackupType, workloadIdentityType))
+		}
+	case backup.CredentialsRef.APIVersion == corev1.SchemeGroupVersion.String() &&
+		backup.CredentialsRef.Kind == "Secret":
+		_, err := v.secretLister.Secrets(backup.CredentialsRef.Namespace).Get(backup.CredentialsRef.Name)
+		if err != nil {
+			return apierrors.NewInternalError(err)
+		}
+	default:
+		return apierrors.NewBadRequest("unsupported credentials reference: backup config is referencing neither a Secret nor a WorkloadIdentity")
 	}
 
-	workloadIdentity, err := v.workloadIdentityLister.WorkloadIdentities(seed.Spec.Backup.CredentialsRef.Namespace).Get(seed.Spec.Backup.CredentialsRef.Name)
-	if err != nil {
-		return apierrors.NewInternalError(err)
+	if oldSeed != nil && oldSeed.Spec.Backup != nil {
+		oldBackup := oldSeed.Spec.Backup
+
+		// If credentials reference has not changed, we can skip the authorization check
+		if oldBackup.CredentialsRef.Kind == backup.CredentialsRef.Kind &&
+			oldBackup.CredentialsRef.APIVersion == backup.CredentialsRef.APIVersion &&
+			oldBackup.CredentialsRef.Name == backup.CredentialsRef.Name &&
+			oldBackup.CredentialsRef.Namespace == backup.CredentialsRef.Namespace {
```

Comment:
Can this be simplified to using `"k8s.io/apimachinery/pkg/api/equality.Semantic.DeepEqual"`?
```suggestion
		if equality.Semantic.DeepEqual(oldBackup, backup) { // "k8s.io/apimachinery/pkg/api/equality"
```